packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -1101,6 +1101,8 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
       return undefined;
     }
 
+    this.node.addDependency(props.vpc.selectSubnets(props.vpcSubnets).internetConnectivityEstablished);
```

Comment:
Can we call `selectSubnets` once, and use the result for both the `subnetIds` and the dependency? 

In other words, add the dependnecy where `selectSubnets` is already being called?